### PR TITLE
Refactor nested conditions with guard clauses [WIP]

### PR DESCRIPTION
**Pull request by andela-osule**
Please review.

Thanks.

### CODE REVIEW

**Comment by ceemion** [comment]
Are you through with the refactoring?

**Comment by andela-osule** [comment]
Oh yes! I am. The change I made with setting the lookup hash as a class variable was an optimization for performance.

File: lib/locations_ng/city.rb [hunk 1]
```diff
@@ -1,31 +1,32 @@
...
         end
+
+        {message: "No cities found for '#{state}'", status: 404}
       end
 
       private
-
       def load_cities
-        YAML.load(File.read(files_location 'cities'))
+        @@all_cities ||= YAML.load(File.read(files_location 'cities'))
```
Comment (ceemion) [inline lib/locations_ng/city.rb]: Oh, I had thought you refactored this into a separate class.

**Comment by ceemion** [comment]
Good to go!

File: lib/locations_ng/city.rb (hunk 1 above)
Comment (andela-osule) [inline lib/locations_ng/city.rb]: Good idea. I will move this to a separate class in the weekend and see if there are any performance gains. I will let you know how that goes.

**Pull request merged**